okhttp/src/main/java/io/grpc/okhttp/OkHttpProtocolNegotiator.java

```diff
@@ -132,6 +136,29 @@ public String getSelectedProtocol(SSLSocket socket) {
     // setNpnProtocol(byte[])
     private static final OptionalMethod<Socket> SET_NPN_PROTOCOLS =
         new OptionalMethod<>(null, "setNpnProtocols", byte[].class);
+    // setApplicationProtocols(String[])
+    private static final OptionalMethod<SSLParameters> SET_APPLICATION_PROTOCOLS =
+        new OptionalMethod<>(null, "setApplicationProtocols", String[].class);
+    // getApplicationProtocol()
+    private static final OptionalMethod<SSLSocket> GET_APPLICATION_PROTOCOL =
+        new OptionalMethod<>(String.class, "getApplicationProtocol");
+    // Non-null on Android 10.0+.
+    // SSLSockets.isSupportedSocket(SSLSocket)
+    private static Method sslSocketsIsSupportedSocket;
```

Comment:
final? Ditto below. (Yes, that is more annoying for the static constructor, but we should probably do it anyway.)

Reply:
Done.